units/systemd-boot-clear-sysfail.service

```diff
@@ -0,0 +1,28 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Clear SysFail entry if the boot is successful
+Documentation=man:systemd-boot-sysfail.service(8)
+
+DefaultDependencies=no
+After=local-fs.target systemd-random-seed.service
+Conflicts=shutdown.target
+Before=sysinit.target
+
+# If LoaderSysFailReason is set we should not clear SysFail entry
+ConditionPathExists=!/sys/firmware/efi/efivars/LoaderSysFailReason-4a67b082-0a4c-41cf-b6c7-440b29bb8c4f
```

Comment:
@poettering Having this conditional prevents `LoaderEntrySysFail` to be cleared in case system failure occurs (and error is reported in `LoaderSysFailReason`). However this leads to failed integration tests:
```
[FAILED] Failed to start systemd-boot-clear-sysfail.service.
```
I was thinking about adding a simple `if` bash condition instead:
```
ExecStart=sh -c "if [ ! -f $/sys/firmware/efi/efivars/LoaderSysFailReason-4a67b082-0a4c-41cf-b6c7-440b29bb8c4f ]; then exit 0; else bootctl set-sysfail ""; fi"
```

Any advise how that can be addressed in a better way?

Reply:
Turned out that the root cause of the integration test failure was different